pkg/diagnose/firewall.go

```diff
@@ -288,3 +304,32 @@ func getTargetPort(submariner *v1alpha1.Submariner, endpoint *subv1.Endpoint, tg
 		return 0, fmt.Errorf("could not determine the target port for cable driver %q", endpoint.Spec.Backend)
 	}
 }
+
+func getLbNodePort(clusterInfo *cluster.Info, endpoint *subv1.Endpoint, tgtport TargetPort, status reporter.Interface) (int32, bool) {
+	var lbNodePort int32
+
+	usingLoadBalancer, _ := endpoint.Spec.GetBackendBool(subv1.UsingLoadBalancer, nil)
+
+	if usingLoadBalancer == nil || !*usingLoadBalancer {
+		return lbNodePort, false
+	}
+
+	portName := encapsPortName
+	if tgtport == NatDiscoveryPort {
+		portName = nattPortName
+	}
+
+	svc, err := clusterInfo.ClientProducer.ForKubernetes().CoreV1().Services(endpoint.GetNamespace()).Get(
+		context.TODO(), loadBalancerName, metav1.GetOptions{})
+	if err == nil {
+		for _, port := range svc.Spec.Ports {
+			if port.Name == portName {
+				return port.NodePort, true
+			}
+		}
+	} else {
+		status.Failure(fmt.Sprintf("Error retrieving gateway LB service: %v", err))
```

Comment:
This is an API failure so we should propagate the error to the caller so it can abort in order to be consistent with the rest of the code base. However we should not treat a `NotFound` error as a failure.